src/web/app/components/question-edit-form/question-edit-form.component.ts

```diff
@@ -192,9 +192,16 @@ export class QuestionEditFormComponent implements OnInit {
    * Change the {@code giverType} and {@code recipientType} and reset the visibility settings.
    */
   changeGiverRecipientType(giverType: FeedbackParticipantType, recipientType: FeedbackParticipantType): void {
+    // check if current recipientType is allowed for giverType,
+    // if not, set default recipientType to the first allowed type as default.
+    const allowedRecipientTypes: FeedbackParticipantType[] | undefined = this.allowedFeedbackPaths.get(giverType);
+    let newRecipientType: FeedbackParticipantType = recipientType;
+    if (allowedRecipientTypes && allowedRecipientTypes.indexOf(recipientType) === -1) {
+      newRecipientType = allowedRecipientTypes[0];
+    }
     this.triggerModelChangeBatch({
       giverType,
-      recipientType,
+      newRecipientType,
```

Comment:
This won't work. Please take a look at https://www.sitepoint.com/es6-enhanced-object-literals/ `Object Initialization From Variables`